kairon/api/app/routers/bot/analytics.py

```
@@ -0,0 +1,78 @@
+    current_user: User = Security(Authentication.get_current_user_and_bot)
+):
+    AnalyticsPipelineEvent(current_user.get_bot(),current_user.get_user()).delete_schedule(event_id)
+
+    return Response(message="Event deleted")
+
+
+@router.put("/events/{event_id}", response_model=Response)
+async def update_pipeline_event(
+    event_id: str,
+    request: AnalyticsPipelineEventRequest,
+    current_user: User = Security(Authentication.get_current_user_and_bot)
```

Ruff (0.14.7)

Do not perform function call `Security` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)